dask/utils.py

```diff
@@ -1268,6 +1268,12 @@ def is_arraylike(x):
 
 def is_dataframe_like(df):
     """Looks like a Pandas DataFrame"""
+    if (df.__class__.__module__, df.__class__.__name__) == (
+        "pandas.core.frame",
+        "DataFrame",
+    ):
+        # fast exec for most likely input
```

Comment:
+1

Reply:
Amusing on this one, I had initially put `isinstance(df, pd.DataFrame)`, and got no name error because pycharm automatically added the import at the top of the file without my realising. It failed in CI because this module shold be importable without pandas.